src/main/java/com/movie_collection/dal/interfaces/ICategoryDAO.java

```diff
@@ -0,0 +1,27 @@
+package com.movie_collection.dal.interfaces;
+
+import com.movie_collection.be.Category;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public interface ICategoryDAO {
+    /**
+     * Retrieves all Categories from the database and store into a list
+     * @return a list of all the Categories from database
+     */
+    List<Category> getAllCategories() throws SQLException;
+
+    /**
+     * Saves the new information about a new Caterogy into the database
+     * @param category the Category to be added
+     */
+    void addCategory(Category category) throws SQLException;
```

Comment:
again here why are we returning void how can we in upper layers check whenever it was even successful or not ?

Reply:
My bad merging unfinished branch

https://github.com/TomassSimko/Private-Movie-Collection/pull/9/commits/83fb3dcba6ee3d26f594102bb50653cc03ca6b50